### DISCUSSION TITLE
Is it possible to change the pinout of your program?

### DISCUSSION THREAD

**Discussion by Roxcreed** [awawa-dev/HyperSerialESP32#3] [Q&A]
Hello everyone, 
I'm trying to make an ambilight on my monitor (my pc is on windows 11). I have a QuinLED-Dig-Quad v3.1 (and a quinled-ESP32 ), my leds are "APA102-type" (HD107S to be more accurate, still 4 wire) and a 5V power supply. On this board, the outpu available are GPIO 16, 3, 1, 4. As GPIO 3,1 are used for RX TX, it lets me only GPIO 16, 4 to use my LEDs

I have installed HyperSerialESP32 with esptool after editing it. My settings on your program are : 

"bool      skipFirstLed = false;    // if set the first led in the strip will be set to black (for level shifters)
int       serialSpeed = 2000000;   // serial port speed
// data output on the hardware SPI pins: MOSI for data, and MSCLK or CLK for clock
#define   is_WS2801                // for WS2801 use NeoPixelBus<NeoRgbFeature, NeoWs2801Spi2MhzMethod>, otherwise comment it with //, choose NeoxxxFeature that fits you
#define   DATA_PIN   16             // PIN: data output for LED strip
#define   CLOCK_PIN  4             // PIN: clock output for LED strip"

On HyperHDR, my LEDs are detected on COM3 but there is nothing happening. LEDs stay black. Any idea of what's wrong ?

**Comment by awawa-dev** [comment]
Hi
Try to comment out that line:
```#define is_WS2801  ....```
becomes
```// #define is_WS2801 ...```
Or just remove it

**Reply by Roxcreed** [reply]
Thanks for your quick answer, 

I'm a bit dumb, you write it on the main page but I didn't understand it that way. 
So I just try to comment that line (// #define is_WS2801), upload it with esptool, reboot everything but the LEDs stay black. By the way, the board has a level shifter.

Here is my config on HyperHDR : 
![image](https://user-images.githubusercontent.com/113352863/189948823-56fb5724-f6c8-4e76-aa65-5de3a6635542.png)

So do you think it's a board issue or LED not supported ? or something else ?

**Reply by awawa-dev** [reply]
That LED strip should be supported, but your serial port chip on ESP not necessarily supports 2Mb speed.
In HyperSerialESP32:
int serialSpeed = 2000000; 
to 
int serialSpeed = 512000; 
And upload it.

Also in HyperHDR adalight configuration change also that speed to 512000. Then test it.

**Reply by awawa-dev** [reply]
Also please verify HyperHDR logs and search for errors. If you are playing with Arduino Studio then HyperHDR may not open that com port at the same time.

**Reply by awawa-dev** [reply]
> So I just try to comment that line (// #define is_WS2801), upload it with esptool,

Make sure you compile it first ;) you don't need esptool later. Arduino Studio (Arduino Software IDE) has built-in up-loader and can do it automatically.

**Reply by Roxcreed** [reply]
I try to lower the baudrate, It seems to work now (seems because I think I have an issue of power injection that I will solve later then edit this message). 

It's weird that the chip can't handle a high baudrate. Quinled ESP32 use esp32-wroom-32e chip. But I don't know where I can read the max speed on datasheet : https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32e_esp32-wroom-32ue_datasheet_en.pdf 

What ESP32 are you using to achieve 2Mb speed ? 

In any case, thanks for your help

**Reply by awawa-dev** [reply]
Since 512000 speed is confirmed to work, you can also test a bit higher speed 921600 and 1500000 now. Some users and I have achieved very good results with the CH9102x (even 4000000 speed should work).

**Reply by awawa-dev** [reply]
> Quinled ESP32 use esp32-wroom-32e chip. 

It's a ESP32 main chip. But on the board should be a second one that handles serial port communication. It should have markings e.g. ch340g, cp2102, cp2104, CH9102x ...

**Reply by Roxcreed** [reply]
Hello, sorry for the late answer

So I try higher baudrate than 500 000 and it flicker like hell or no response. The chip handling serial port is CH340C (C not G), and I can't find spec. So I guess it's not the same as CH340G which can handle higher baudarate ? 

Anyway, I get this weird behavior where there is lot of latency. (maybe not the best video but I think you will see the latency)
https://youtu.be/4laKpHChlOQ

I get back to WLED as it is smoother at the same baudrate actually. I just wanted to show you this weird behavior

Thanks for all your help.

**Reply by awawa-dev** [reply]
Flickering must have some other cause not related to the communications problem since AWA protocol has protection for the data integrity (simply adalight doesn't have it). Like some conflict or resources. CH340C should reach 2Mb. But I don't know what modified sketch you are using. You have also some debugging info that can be read from serial port when HyperHDR is not running (but it ran before that).
